nobo.py

```diff
@@ -387,40 +387,68 @@ def response_handler(self, r):
             dicti = dict(zip(self.API.STRUCT_KEYS_COMPONENT, r[1:]))
             self.components[dicti['serial']] = dicti
             logging.info('added/updated component: %s', dicti['name'])
 
-        elif r[0] in [self.API.RESPONSE_WEEK_PROFILE_INFO, self.API.RESPONSE_UPDATE_V02]:
+        elif r[0] in [self.API.RESPONSE_WEEK_PROFILE_INFO, self.API.RESPONSE_ADD_WEEK_PROFILE, self.API.RESPONSE_UPDATE_WEEK_PROFILE]:
             dicti = dict(zip(self.API.STRUCT_KEYS_WEEK_PROFILE, r[1:]))
             dicti['profile'] = r[-1].split(',')
             self.week_profiles[dicti['week_profile_id']] = dicti
             logging.info('added/updated week profile: %s', dicti['name'])
 
-        elif r[0] in [self.API.RESPONSE_OVERRIDE_INFO, self.API.RESPONSE_ADD_B03]:
+        elif r[0] in [self.API.RESPONSE_OVERRIDE_INFO, self.API.RESPONSE_ADD_OVERRIDE]:
             dicti = dict(zip(self.API.STRUCT_KEYS_OVERRIDE, r[1:]))
             self.overrides[dicti['override_id']] = dicti
             logging.info('added/updated override: id %s', dicti['override_id'])
 
-        elif r[0] in [self.API.RESPONSE_STATIC_INFO, self.API.RESPONSE_UPDATE_V03]:
+        elif r[0] in [self.API.RESPONSE_HUB_INFO, self.API.RESPONSE_UPDATE_HUB_INFO]:
             self.hub_info = dict(zip(self.API.STRUCT_KEYS_HUB, r[1:]))
             logging.info('updated hub info: %s', self.hub_info)
-            if r[0] == self.API.RESPONSE_STATIC_INFO:
+            if r[0] == self.API.RESPONSE_HUB_INFO:
                 self.socket_received_all_info.set()
 
-        #TODO: Add all the other response_remove commands
-        elif r[0] == self.API.RESPONSE_REMOVE_S03:
+        # The removed info messages 
+        elif r[0] == self.API.RESPONSE_REMOVE_ZONE:
+            dicti = dict(zip(self.API.STRUCT_KEYS_ZONE, r[1:]))
+            popped_zone = self.zones.pop(dicti['zone_id'], None)
+            logging.info('removed zone: %s', dicti['name'])
+
+        elif r[0] == self.API.RESPONSE_REMOVE_COMPONENT:
+            dicti = dict(zip(self.API.STRUCT_KEYS_COMPONENT, r[1:]))
+            popped_component = self.components.pop(dicti['serial'], None)
+            logging.info('removed component: %s', dicti['name'])
+
+        elif r[0] == self.API.RESPONSE_REMOVE_WEEK_PROFILE:
+            dicti = dict(zip(self.API.STRUCT_KEYS_WEEK_PROFILE, r[1:]))
+            popped_profile = self.week_profiles.pop(dicti['week_profile_id'], None)
+            logging.info('removed week profile: %s', dicti['name'])
+
+        elif r[0] == self.API.RESPONSE_REMOVE_OVERRIDE:
             dicti = dict(zip(self.API.STRUCT_KEYS_OVERRIDE, r[1:]))
             popped_override = self.overrides.pop(dicti['override_id'], None)
             logging.info('removed override: id%s', dicti['override_id'])
 
+        # Component temperature data
+        elif r[0] == self.API.RESPONSE_COMPONENT_TEMP:
```

Comment:
I have never seen this message from my thermostats. I did see the temperatures when connecting to the serial ports though. Have you seen these reports?

Reply:
I do not have any thermostats with temperature reporting :) I just added it from description for Y02